### ISSUE TITLE
On trial detail pages, add the section for register entries

### ISSUE THREAD

**Issue by pwalsh** [opentrials/opentrials#58]
# Description

The mocks for the trial detail pages have a section for registry entries of the trial [see here](http://opentrials.github.io/mockups/trial.html).

This section is important as it shows users the core information that we built our "ideal trial object" from. This section also should allow the user to see the record from the register as we collected it, and, follow a link to the original page or API call that returns that information.

# Tasks

* [ ] Implement the section of the page for registers
* [ ] Clicking on the element for each register should open a simple modal, which, in the simplest way possible, renders the data of the trial record from that register as a list of key: value
* [ ] Above the key: value list, have a template snippet like:
  - "This record was cached from {REGISTER_NAME} on {DATE_OF_ACQUISITION}. See the source data at this link: {LINK_TO_SOURCE_PAGE}"

**Comment by vitorbaptista** [comment]
@roll Do we have this in data in the DB already?

**Comment by roll** [comment]
We have `records` table linked to `sources` but we need to add `url` and `date` to implement this. Key-value dict for record is `data` field.

**Comment by roll** [comment]
Also because we need to show some free data structures (free JSON) I've checked for tools for converting it to html tables:
https://github.com/moappi/json2html
https://github.com/padolsey-archive/prettyprint.js
May be it will be useful once.

**Comment by roll** [comment]
@pwalsh Great thing that data providing work starts to meet with concrete user stories here. 

So a question - `records.data` should contains all data from warehouse item (https://github.com/opentrials/collectors/blob/master/docs/warehouse/warehouse.md) like 100+ fields for `euctr` or some useful subset?

**Comment by pwalsh** [comment]
@roll all of it.

**Comment by roll** [comment]
@pwalsh ok
@vitorbaptista could we have database migration you need for this feature before database update will be started? So we will be able to close this issue soon.

**Comment by pwalsh** [comment]
@roll yes, please as part of migration to update database as this is extremely critical information.

**Comment by vitorbaptista** [comment]
@roll I think we should use dedicated fields as much as possible :+1: Is this `date` field related to when the data was imported, or when it was created in the registry?

Not directly related to this, but is it possible to use a dedicated field for the source's ID as well (e.g. `nct_id`)? It would make it easier to display on the frontend.

**Comment by vitorbaptista** [comment]
@pwalsh instead of opening a modal with the source data, what do you think of something like this:

![screenshot from 2016-04-06 15 27 20](https://cloud.githubusercontent.com/assets/76945/14320322/869fdec4-fc0c-11e5-9628-363004cafcb8.png)

My reasoning was that knowing when the data was last updated is important enough not to be hidden behind a click. It also keeps the interface a bit more consistent, considering that after #57 the boxes in conditions and treatments, which are visually identical as the ones for registries, will link to a separate page.

**Comment by pwalsh** [comment]
@vitorbaptista yes great, big +1 on that solution, but one caveat - you are linking out to the actual source, which is good, but we are losing the ability to see our cached version of the source, which is important I think.

Thoughts on this?

**Comment by roll** [comment]
> Not directly related to this, but is it possible to use a dedicated field for the source's ID as well (e.g. nct_id)? It would make it easier to display on the frontend.

I think almost everything is possible with `processors` - please just raise a PR with migration you need for frontend. That's workflow I was talking about in Gitter - when you lead those migrations (and I think we need some) based on better knowledge of clients (frontend) needs.

**Comment by vitorbaptista** [comment]
@roll Great, I've created a new issue on https://github.com/opentrials/opentrials/issues/78 about it

@pwalsh I was thinking on linking to the actual JSON of our source. If that's too technical, we could create a separate page displaying it. Basically what would be in the modal, but in a separate page instead.

**Comment by pwalsh** [comment]
Separate page is fine...

**Comment by pwalsh** [comment]
JSON source is not though :)

**Comment by roll** [comment]
Take a look on links I've posted above - json2htmltable

ср, 06 апр 2016 г., 19:35 Paul Walsh <notifications@github.com>:

> JSON source is not though :)
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/opentrials/opentrials/issues/58#issuecomment-206454068>
>
-- 
Best regards,
Evgeny Karev

**Comment by vitorbaptista** [comment]
Tasks:

- [ ] Add sources endpoint `/trials/TRIAL_ID/sources/SOURCE_ID`;
- [ ] Add basic sources metadata to `/trials/TRIAL_ID` (source name, URL, and maybe ID);
- [ ] Write page to display the data extracted from the source (maybe using `json2htmltable` or something similar)
- [ ] Add sources list to the Trial details page
